File: wheel_factorization.rs
fn wheel_factorization(mut num: i32, primes: &Vec<i32>) -> Vec<i32> {
    let mut factors = Vec::new();
    for &prime in primes {
        if prime * prime > num { break }
        if num % prime == 0 {
            factors.push(prime);
            while num % prime == 0 {
                num /= prime;
            }
        }
    }
    if num > 1 { factors.push(num) }
    factors
}

File: wheel_factorization.cpp
vector<int> wheelFactorization(int num, vector<int>& primes) {
    vector<int> factors;
    for (int prime : primes) {
        if (prime * prime > num) break;
        if (num % prime == 0) {
            factors.push_back(prime);
            while (num % prime == 0) num /= prime;
        }
    }
    if (num > 1) factors.push_back(num);
    return factors;
}
